### DISCUSSION TITLE
App Builder on-prem - authentication and Internet connectivity considerations

### DISCUSSION THREAD

**Discussion by turbobobbytraykov** [IgniteUI/app-builder#40] [General]
Hi App Builder Team,

I'm reading through the [product overview](https://www.infragistics.com/products/appbuilder/app-builder-overview) and [relevant docs page](https://www.appbuilder.dev/help/on-prem-prerequisites-and-installation) and my company is evaluating it.
Unfortunately I am not able to find any details on the questions I've listed below so your response will be greatly appreciated.

We are relying on on-prem and in some cases air-gapped environments.
- Does my company need to pay per internal user of the App Builder on-prem instance?
- What are the requirements for potentially hooking an App Builder on-premise instance to an internal authentication provider?
- Is Internet access required in order to run an on-prem instance?
  - If yes, can we get a list of URLs and ports that are necessary and can we use a proxy on the App Builder's back-end settings or should we do so manually via a jump host?
  
 Thank you in advance for all the details that you can share!

**Comment by zdrawku** [comment]
Hey there and apologies for the late response, I’ll do my best to fully answer your questions.

1. **Does my company need to pay per internal user of the App Builder on-prem instance?**
As stated on our website, we recommend reaching out to your regional sales representative for personalized assistance. An App Builder on-premise subscription typically includes 5 users by default, with options to add more based on customer requirements. Pricing details are not publicly disclosed and are always managed through a sales process. You can find additional guidance on our pricing page - https://www.appbuilder.dev/pricing.

2. **What are the requirements for potentially hooking an App Builder on-premise instance to an internal authentication provider?**
To configure authentication for an App Builder on-premise instance, you’ll need an OpenID Connect (OIDC) server that supports OAuth 2.0 Authorization Code Flow with PKCE. I will have that logged as a task, in order to provide a document detailing how to achieve that as we dont have it publically available. The document will cover steps like:
- Providing `OIDC` client settings.
- Required configuration settings for App Builder.
- Additional options such as `oidc_scope` and `oidc_redirect_uri`.
3. **Is Internet access required in order to run an on-prem instance?** 
No, Internet access is not required. Te majority of the services and features dependent on Internet connectivity are disabled in the on-premise solution by default. And most of these can be enabled through environment configuration when running the Docker image. Again, this is something that is not publicly available and we will expose as a document.
4. **If yes, can we get a list of URLs and ports that are necessary and can we use a proxy on the App Builder's back-end settings or should we do so manually via a jump host?**
We can provide a detailed list of external references for whitelisting in documentation format (e.g., the urls that reffer to library component documentation and video tutorials). Currently, proxy configuration is not available within the App Builder's settings; users are responsible for configuring proxies within their network.

**Action Items out of your post that I will plan for the upcoming itterations:**
1. Create and share public documentation for setting up custom authentication providers. - [Work Item](https://github.com/IgniteUI/app-builder/issues/41)
2. Publish a document outlining available configuration flags (e.g., `docker run --restart always -p 80:5000 -e AuthSettings__ClientId="1234-4657-00"`) - [Work item](https://github.com/IgniteUI/app-builder/issues/42).
3. Provide a documentation-based list of external references if required for whitelisting - [Work Item](https://github.com/IgniteUI/app-builder/issues/43)

Thank you, @turbobobbytraykov, for your dedication and effort in helping us improve our platform 🙌

**Comment by zdrawku** [comment]
@turbobobbytraykov, we now have the downloadable links for the Desktop AB app and soon for the On-premise file. And we also have the missing topics explaining everything from above:

Auth with OpenID Connect - https://www.appbuilder.dev/help/on-premises/auth-with-openid-connect-o-auth
Configuration flags for On-prem and SDK - https://www.appbuilder.dev/help/on-premises/configuration-flags
Whitelisting external resources - https://www.appbuilder.dev/help/on-premises/external-references-for-whitelisting
